src/main/java/com/seatapp/domain/Reservation.java

```diff
@@ -132,4 +152,20 @@ public boolean checkIfStartTimeIsBeforeEndTime(final LocalDateTime
     public boolean checkIfDateIsAfterNow(final LocalDateTime startTimeNew) {
         return startTimeNew.isAfter(LocalDateTime.now());
     }
+
+    /**
+     * Changes the checked in status of a reservation.
+     * @param username username of the person
+     *                wanting to check in.
+     */
+    public void checkIn(final String username) {
+        if (isCheckedIn()) {
+            throw new IllegalArgumentException(
+                    "This reservation is already checked in.");
+        } else if (!username.equals(user.getUsername())) {
```

Comment:
The `else` here is not needed because when an error is thrown it won't do anything after the error.